fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -1288,8 +1288,13 @@ Reference<ILogSystem::IPeekCursor> TagPartitionedLogSystem::peekSingle(UID dbgid
 Reference<ILogSystem::IPeekCursor> TagPartitionedLogSystem::peekLogRouter(UID dbgid,
                                                                           Version begin,
                                                                           Tag tag,
```

Comment:
I see that "getPeekEnd()" returns "recoverAt" (if set) - the invoker of "peekLogRouter()" would also set "end" to "recoverAt", right? If so, when is "getPeekEnd()" going to be different from "end"?

Reply:
This PR modifies the code so that on recovery, the invoker to `peekLogRouter` will set "end" to `recoverAt`. Prior to this PR, the invoker to "peekLogRouter()" (on recovery) set `end` to std::numeric_limits<>, i.e. infinity.

Reply:
This cursor is invoked both during recovery and during normal operation. When called during recovery, we must send `end` of the previous generation. That did not exist prior to the PR, and is now sent in a new parameter. When called during normal operation, `recoverAt` is not present and therefore getPeekEnd sets `recoverAt` to `std::numeric_limits<Version>::max()`  I agree this is unclear (as `end`  can represent both a recovery version or `std::numeric_limits<Version>::max()`  and will investigate refactoring to make it clearer.

Reply:
If you can refactor it to make it better that's fine, but I meant that the log router is already invoking this with "end" set to "recoverAt" (and this PR is making the recovering version available to the log router) so why would "end" be not present when called from a log router/a peeking log server during recovery?

Reply:
The PR makes a change such that during recovery, `end` will be set to be `recoverAt`. If it is not set, we are not in recovery, and `end` will be `std::numeric_limits<Version>::max()`. We cannot call `getPeekEnd` where we used to, because it won't have `recoverAt`. This is because when the log router starts for recovery (from `newEpoch`), a new log system is still being built and has not yet been sent via cstate to all workers. Some refactoring is possible, but calling `getPeekEnd` as we used to is not possible.

I don't completely follow what you are saying, so if this answer does not make sense lets discuss offline.